frontend/src/component/environments/EnvironmentTable/EnvironmentActionCell/EnvironmentCloneModal/EnvironmentCloneModal.tsx

```diff
@@ -0,0 +1,385 @@
+        }/api/admin/environments/${environment.name}/clone' \\
+    --header 'Authorization: INSERT_API_KEY' \\
+    --header 'Content-Type: application/json' \\
+    --data-raw '${JSON.stringify(getCloneEnvironmentPayload(), undefined, 2)}'`;
+    };
+
+    const isNameNotEmpty = (name: string) => name.length;
+    const isNameUnique = (name: string) =>
+        !environments?.some(environment => environment.name === name);
+    const isValid =
+        isNameNotEmpty(name) && isNameUnique(name) && tokenProjects.length;
+
+    const onSetName = (name: string) => {
+        clearError(ErrorField.NAME);
+        if (!isNameUnique(name)) {
+            setError(
+                ErrorField.NAME,
+                'An environment with that name already exists.'
+            );
+        }
+        setName(name);
+    };
+
+    const selectableProjects = allProjects.map(project => ({
+        value: project.id,
+        label: project.name,
+    }));
+
+    return (
+        <SidebarModal
+            open={open}
+            onClose={() => {
+                setOpen(false);
+            }}
+            label={`Clone ${environment.name} environment`}
+        >
+            <FormTemplate
+                loading={loading}
+                modal
+                title={`Clone ${environment.name} environment`}
+                description="Cloning an environment will clone all feature toggles and their configuration (activation strategies, segments, status, etc) into a new environment."
+                documentationLink="https://docs.getunleash.io/user_guide/environments#cloning-environments"
+                documentationLinkLabel="Cloning environments documentation"
+                formatApiCode={formatApiCode}
+            >
+                <StyledForm onSubmit={handleSubmit}>
+                    <div>
+                        <StyledInputDescription>
+                            What is your environment name? (Can't be changed
+                            later)
+                        </StyledInputDescription>
+                        <StyledInput
+                            autoFocus
+                            label="Environment name"
+                            error={Boolean(errors.name)}
+                            errorText={errors.name}
+                            value={name}
+                            onChange={e => onSetName(e.target.value)}
+                            required
+                        />
+                        <StyledInputDescription>
+                            What type of environment do you want to create?
+                        </StyledInputDescription>
+                        <EnvironmentTypeSelector
+                            onChange={e => setType(e.currentTarget.value)}
+                            value={type}
+                        />
+                        <StyledInputDescription>
+                            Select for which projects to clone the environment
```

Comment:
Can't do a suggest change because two lines and all but this reads a little awkward to me. What about

"Select which projects you want to clone the environment configuration in" ?